streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -415,6 +422,38 @@ protected void maybeClean() {
         }
     }
 
+    /**
+     * Returns ids of tasks that were being executed before the rebalance.
+     */
+    public Set<TaskId> prevTasks() {
+        return prevTasks;
+    }
+
+    /**
+     * Returns ids of tasks whose states are kept on the local storage.
+     */
+    public Set<TaskId> cachedTasks() {
```

Comment:
Add some comments: a client could contain some inactive tasks whose states are still kept on the local storage in the following scenarios:

1) the client is actively maintaining standby tasks by maintaining their states from the change log.
2) the client has just got some tasks migrated out of itself to other clients while these task states have not been cleaned up yet (this can happen in a rolling bounce upgrade, for example).